### PR TITLE
SymDB: hot-load follow-ups — error boundary, race guard, deterministic tests, RBS type

### PR DESCRIPTION
**Pull request by p-datadog**
**What does this PR do?**

Five follow-ups to PR #5697 (TracePoint :class hot-load hook for incremental class extraction):

1. **Error boundary around the hot-load TracePoint callback.** The `:class` TracePoint fires inside the customer's `class Foo; ... end` body. An exception escaping the callback propagates *into the class definition itself* — verified empirically: a `raise` inside a `:class` TracePoint backtraces through `<class:CustomerClass>` and surfaces at the customer's `class` keyword, breaking their class load. The earlier `MODULE_SINGLETON_CLASS_PRED` dispatch closed one specific raise source (user-overridden `singleton_class?`); this commit closes the general case with a rescue around the callback body that matches the sibling pattern in `start_upload`/`scheduler_loop`/`extract_and_upload`: log at debug, report via telemetry, allow the customer's class load to proceed.

2. **`enqueue_hot_load` race guard.** `TracePoint#disable` does not wait for in-flight callbacks. A `:class` event firing concurrently with `stop_upload` could reach `enqueue_hot_load` after the hook had been torn down and re-arm the scheduler — contradicting `stop_upload`'s contract. Now bails out when `@hot_load_tracepoint` is nil. The buffer push above the guard is harmless because the next `start_upload` runs `extract_all` (`@initial_extraction_done` was reset), which clears the buffer first.

3. **`Components#after_fork` spec.** Added a dedicated `#after_fork` describe block to `components_spec.rb` covering the `symbol_database&.after_fork!` wiring — both the dispatch path and the nil-safety case.

4. **Three sleeps replaced with deterministic waits in `component_spec.rb`:**
   - "clears scheduler thread reference and pending schedule" — `@scheduler_thread` is assigned synchronously inside `start_upload`'s mutex via `ensure_scheduler_thread`, so the sleep was never needed.
   - "leaves a child Component able to start_upload normally after fork-state reset" — replaced with `wait_for_idle`.
   - "stop_upload prevents a post-stop class definition from triggering extraction" — removed entirely; the `@scheduled_at` nil check is the structural proof (only `enqueue_hot_load` sets it after `stop_upload`, and `enqueue_hot_load` only runs if the TracePoint fired).

5. **RBS type fix.** `EXTRACT_DEBOUNCE_INTERVAL` is declared `Integer` but participates in arithmetic with `Float` clock values and is stubbed to `0.05` in tests. Changed to `::Numeric`.

**Motivation:**

Surfaced during review of #5697 after merge. Items 1 and 2 are behavioral; the rest are test/type hygiene.

Item 1 is the highest-impact: without the rescue, any unexpected raise inside the callback (mutex edge case, user-set `Time.now_provider` raising, future code change in `enqueue_hot_load`) propagates into customer code at `class Foo; ... end`. Item 2 closes a documented race window in which symdb can re-arm the scheduler after the customer disables it via remote config.

**Change log entry**

None.

**Additional Notes:**

- Item 1 verification (Ruby 3.2.3): raising `CustomerError` inside a `TracePoint :class` callback produced a backtrace of `block in <main>` → `<class:CustomerClass>` → `<main>`. The exception was caught by a `rescue` placed around the surrounding `class CustomerClass; end` block — confirming the callback's raise propagates through the class definition body and into customer scope.

**How to test the change?**

- `bundle exec rspec spec/datadog/symbol_database/component_spec.rb` — 51 examples, 0 failures. New specs:
  - `rescues exceptions raised inside the :class hook so customer class loads do not break` (item 1)
  - `enqueue_hot_load called after stop_upload does not re-arm the scheduler` (item 2)
- `bundle exec rspec spec/datadog/core/configuration/components_spec.rb` — passes including the two new `#after_fork` examples (item 3).
- `bundle exec steep check lib/datadog/symbol_database/component.rb` — clean.

### CODE REVIEW

**Comment by p-datadog** [comment]
@codex review

File: lib/datadog/symbol_database/component.rb [hunk 1]
```diff
@@ -540,10 +540,24 @@ def extract_hot_load_buffer
...
+          # raise source (user-overridden singleton_class?); this rescue
+          # closes the general case. Verified: a raise inside the callback
+          # backtraces through `<class:CustomerClass>` in Ruby 3.x.
+          begin
+            mod = tp.self
+            next if MODULE_SINGLETON_CLASS_PRED.bind(mod).call
+            component.send(:enqueue_hot_load, mod)
+          rescue => e
+            logger.debug { "symdb: hot-load hook error: #{e.class}: #{e.message}" }
+            telemetry&.report(e, description: 'symdb: hot-load hook error')
```
[the remark this answers is not in the document]
Comment (p-datadog) [reply lib/datadog/symbol_database/component.rb]: Good catch — the outer rescue can still re-raise if `logger.debug` or `telemetry.report` themselves raise (custom logger throwing, telemetry worker in an unexpected state), which defeats the error-boundary goal.

Fixed in **lib/datadog/symbol_database/component.rb:557** by wrapping the logging/telemetry calls in their own rescue. The inner rescue swallows silently because there is nothing useful to do if error reporting itself fails; a comment explains the silent rescue.

Added a test in **spec/datadog/symbol_database/component_spec.rb** that stubs `raw_logger.debug` to raise and verifies a class definition does not propagate the exception.

Pattern search: searched `TracePoint.new` in `lib/datadog/symbol_database/` and `lib/datadog/di/`. Same shape exists at `lib/datadog/di/code_tracker.rb:194` (logger + telemetry inside the outer rescue handler, no inner rescue). Out of scope for this PR — flagging here in case you want to address it separately.

**Review by Strech — APPROVED** [review]
